user_id.py

```diff
@@ -1,33 +1,23 @@
-"""Use a given name string, a family name string, and a random integer
-between 100 and 999 to create a user_id.
+"""Use two strings, 'given_name' and 'family_name', to create a hash
+value as the user_id.
 """
-import random
 import re
+import hashlib
 import name
 
 def is_valid(user_id_str):
     """Take in a string containing a user ID. Return True if it is
-    valid user ID. Otherwise, return False.
+    valid. Otherwise, return False.
     """
-    if re.search("[a-zA-Z]+[0-9][0-9][0-9]", user_id_str):
-        return True
-    return False
+    return True if re.search("[a-zA-Z0-9]{56}", user_id_str) else False
 
 def create(given_name, family_name):
-    """Take in two strings: a given_name and a family_name. Use the
-    strings and a random integer between 100 and 999 to create a unique
-    user_id. Example using John Smith:
-    jsmith###
+    """Take in two strings: a given_name and a family_name. Validate
+    them and use them to create user ID as a hash value. Return the ID.
     """
     assert name.is_valid(given_name)
-    id_head = given_name[0].lower()
-
     assert name.is_valid(family_name)
-    id_body = family_name.lower()
-
-    id_tail = random.randint(100, 999)
-
-    return id_head + id_body + str(id_tail)
+    return hashlib.sha224(b""+given_name+" "+family_name).hexdigest()
```

Comment:
nice.

TBH, i couldn't tell you the difference between MD5's, sha's, etc... they're just different algorithms.

Hashes are also used to hide passwords as frequently you can't go backwards, or it's very hard to, turn a hash back into it's constituent string value.